conslogging/conslogging.go

```diff
@@ -126,6 +126,13 @@ func (cl ConsoleLogger) WithFailed(isFailed bool) ConsoleLogger {
 	return ret
 }
 
+func (cl ConsoleLogger) WithWriter(w io.Writer) ConsoleLogger {
+	ret := cl.clone()
+	ret.outW = w
+	ret.errW = w
+	return ret
+}
+
```

Comment:
Needed a way to be able to also read logs from tests. You could also use this with `https://pkg.go.dev/io#MultiWriter` to effectively tee logs. Could be useful when implementing truncation schemes later.